script.js

```diff
@@ -1,4 +1,341 @@
+      rankCounter += 1;
+    }
+    // increment suit index to iterate over the next suit
+    suitIndex += 1;
+  }
+  // return the completed card deck
+  return cardDeck;
+};
+
+// generate a random index from 0 (inclusive) to max (exclusive)
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+//   shuffle cards
+var shuffleCards = function (cardDeck) {
+  // loop over the card deck array
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    // swap position of random card and current card in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // retutn the shuffled deck
+  shuffledDeck = cardDeck;
+  return shuffledDeck;
+};
+
+// deal 2 cards each to player and dealer from the shuffled deck
+var drawCard = function (cardDeck) {
+  var numOfCard = 2;
+  var cardCounter = 0;
+  while (cardCounter < numOfCard) {
+    playerHand.push(shuffledDeck.pop());
+    dealerHand.push(shuffledDeck.pop());
+    cardCounter += 1;
+  }
+
+  var dealedCards = `player Hand ${playerHand[0].name} of ${playerHand[0].suit} ${playerHand[1].name} of ${playerHand[1].suit} <br> dealer Hand  ${dealerHand[0].name} of ${dealerHand[0].suit}, ${dealerHand[1].name} of ${dealerHand[1].suit} `;
+  return dealedCards;
+};
+
+// compare player and dealer hand and return result.
+
+var checkResult = function () {
+  var winner = ``;
+  // draw condition
+  if (playersTotalPoints > 21 && dealersTotalPoints > 21) {
+    winner = `its a draw or busted: `;
+  }
+  // draw condition normal
+  else if (playersTotalPoints == dealersTotalPoints) {
+    winner = `its a draw `;
+  }
+  // player win by black jack
+  else if (
+    (playerHand[0].rank == 1 || playerHand[1].rank == 1) &&
+    (playerHand[0].rank >= 10 || playerHand[1].rank >= 10)
+  ) {
+    winner = `Black Jack ${player} wins`;
+  }
+  // dealer win by black jack
+  else if (
+    (dealerHand[0].rank == 1 || dealerHand[1].rank == 1) &&
+    (dealerHand[1].rank >= 10 || dealerHand[1].rank >= 10)
+  ) {
+    winner = `Black Jack ${dealer} wins`;
+  }
+  // player busted
+  else if (playersTotalPoints > 21 && dealersTotalPoints <= 21) {
+    winner = `congrats ${dealer} wins`;
+  }
+  // dealer busted
+  else if (playersTotalPoints <= 21 && dealersTotalPoints > 21) {
+    winner`congrats ${player} wins`;
+  }
+  // winning normal conditions
+  else if (playersTotalPoints > dealersTotalPoints) {
+    winner = ` congrats ${player} wins`;
+  } else if (playersTotalPoints < dealersTotalPoints) {
+    winner = `congrats ${dealer} wins`;
+  }
+  return winner;
+};
+
+// function to print player hand
+
+var printPlayerHand = function () {
```

Comment:
the printPlayerHand and printDealerHand is pretty much identical other than the fact that one is using the playerHand and the other uses the dealerHand. Try passing in the hand as an argument to the function and you'll realise that a single function can handle printing for both the player and the computer